src/client/Microsoft.Identity.Client/Internal/IdToken.cs

```diff
@@ -62,24 +62,34 @@ public string GetUniqueId()
 
         public static IdToken Parse(string idToken)
         {
+            string payload = string.Empty;
+
             if (string.IsNullOrEmpty(idToken))
             {
                 return null;
             }
 
-            string[] idTokenSegments = idToken.Split(new[] { '.' });
-
-            if (idTokenSegments.Length < 2)
-            {
-                throw new MsalClientException(
-                    MsalError.InvalidJwtError,
-                    MsalErrorMessage.IDTokenMustHaveTwoParts);
-            }
-
             try
             {
-                string payload = Base64UrlHelpers.Decode(idTokenSegments[1]);
-                var idTokenClaims = JsonConvert.DeserializeObject<Dictionary<string, object>>(payload);                
+                if (TryParseJSON(idToken, out _))
```

Comment:
This will have a perf impact over existing flows. We should try to first decode the JWT and if that fails we can fallback to JSON parsing.